### ISSUE TITLE
GBA.emu build error

### ISSUE THREAD

**Issue by Josh4all** [Rakashazi/emu-ex-plus-alpha#75]
Building X86 Shared Object
make[1]: Entering directory '/home/josh/Games/emu-ex-plus-alpha/GBA.emu'
Linking target/android-release-15/libs/x86/libmain.so
/home/josh/Android/Sdk/ndk-bundle/toolchains/x86-4.9/prebuilt/linux-x86_64/lib/gcc/i686-linux-android/4.9.x/../../../../i686-linux-android/bin/ld: error: emuframework: no archive symbol table (run ranlib)
/home/josh/Android/Sdk/ndk-bundle/toolchains/x86-4.9/prebuilt/linux-x86_64/lib/gcc/i686-linux-android/4.9.x/../../../../i686-linux-android/bin/ld: error: imagine: no archive symbol table (run ranlib)
/home/josh/Android/Sdk/ndk-bundle/toolchains/x86-4.9/prebuilt/linux-x86_64/lib/gcc/i686-linux-android/4.9.x/../../../../i686-linux-android/bin/ld: error: archive: no archive symbol table (run ranlib)
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
/home/josh/Games/emu-ex-plus-alpha/imagine/make/imagineAppTarget.mk:30: recipe for target 'target/android-release-15/libs/x86/libmain.so' failed
make[1]: *** [target/android-release-15/libs/x86/libmain.so] Error 1
make[1]: Leaving directory '/home/josh/Games/emu-ex-plus-alpha/GBA.emu'
/home/josh/Games/emu-ex-plus-alpha/imagine/make/shortcut/meta-builds/android.mk:334: recipe for target 'android-x86' failed
make: *** [android-x86] Error 2

**Comment by Rakashazi** [comment]
Some versions of binutils (including the one in NDK 12) create broken static archives from LLVM bytecode and I'm not sure of the exact reason yet. Two possible solutions:

1. Update the binutils package on your Linux system (2.26 on Gentoo confirmed to work properly), delete all previously built static archives (libimagine.a, libemuframework.a , dependencies like libarchive.a, etc.), and regenerate them with the updated binutils

2. Disable any LTO usage in the makefiles so no LLVM bytecode is produced, and regenerate all static archives as in solution 1.

**Comment by Josh4all** [comment]
Well, I'm on Ubuntu 16.04 LTS and I'm pretty sure that the binutils is 2.26 :(

**Comment by Rakashazi** [comment]
I'm waiting to test on NDK 13 when it's released before investigating the binutils issue further. In the meantime I added an easier way to turn off LTO by defining LTO_MODE=off when building. Make sure you clean out any .o files already built with LTO first.

**Comment by Rakashazi** [comment]
After testing on NDK 13 beta 1 I found the cause the of the issue. The ar tool isn't loading the LLVMgold.so plugin that's used to handle LLVM bitcode from object files. I modified the Android Makefiles to check if the current toolchain contains a bfd-plugins directory, which is where ar will look for plugins to load, and make a symlink to LLVM's lib path if it's not found. That also explains why it was working with my Gentoo system's binutils because I also had LLVM installed system-wide and the plugin was used automatically.
